framework/src/outputs/png/PNGOutput.C

```diff
@@ -58,34 +66,93 @@ void
 PNGOutput::makeMeshFunc()
 {
 
-  const std::vector<unsigned int> var_nums = {0};
+  if (processor_id() != 0)
+    mooseError("PNGOutput does not currently function in parallel.");
+
+  // By default _use_aux is false.
+  _use_aux = false;
+  // The number assigned to the variable.  Used to build the correct mesh.  Default is 0.
+  unsigned int variable_number = 0;
+
+  // If a specific variable name isn't provided, skip this if statement and use the default
+  // varaible.
+  if (_variable != "None")
+  {
+    if (_problem_ptr->getAuxiliarySystem().hasVariable(_variable))
+    {
+      variable_number =
+          _problem_ptr->getAuxiliarySystem().getVariable(processor_id(), _variable).number();
+      _use_aux = true;
+    }
+
+    else if (_problem_ptr->getNonlinearSystem().hasVariable(_variable))
+      variable_number =
+          _problem_ptr->getNonlinearSystem().getVariable(processor_id(), _variable).number();
+
+    else
+      mooseError("The given input variable '",
```

Comment:
We use paramError() for this case (errors involving input file parameters) it gives contextual error information.